desktop_version/src/editor.cpp

```diff
@@ -3008,7 +3009,11 @@ void editorclass::save(std::string& _path)
         TiXmlElement *edlevelclassElement = new TiXmlElement( "edLevelClass" );
         edlevelclassElement->SetAttribute( "tileset", level[i].tileset);
         edlevelclassElement->SetAttribute(  "tilecol", level[i].tilecol);
-        edlevelclassElement->SetAttribute(  "customtileset", level[i].customtileset);
+		// Since these will be 0 aka the default value
+		// We really don't need this for every single room
+		// Only store these if they are a non zero value!
+		if(level[i].customtileset) edlevelclassElement->SetAttribute(  "customtileset", level[i].customtileset);
+		if(level[i].customspritesheet) edlevelclassElement->SetAttribute(  "customspritesheet", level[i].customspritesheet);
```

Comment:
This is indented with tabs when the surrounding code is indented with spaces.

Reply:
Also, why change it so that these two properties are only written when they're nonzero? Seems unnecessary to me, imo. It requires more unnecessary work on the part of XML parsers such as Ved's.

Reply:
My concern was file size, so that we don't unnecessarily increase the level file size for every property of a room that someone at some point in time adds and which will be the default value in all other cases. Plus we still need to be able to parse Vanilla levels where these don't exist.
But if that is a lesser concern, I'll just make these print every time.

Reply:
> My concern was file size, so that we don't unnecessarily increase the level file size for every property of a room that someone at some point in time adds and which will be the default value in all other cases.

Why? What's wrong with having a larger file size?

> Plus we still need to be able to parse Vanilla levels where these don't exist.

Well, yes, but this code is for writing properties to the level file, not parsing them. Just because we have to sometimes parse level files that don't have these room properties doesn't mean we have to write these room properties only when they're nonzero.